graphql_compiler/compiler/ir_lowering_common/location_renaming.py

```diff
@@ -1,11 +1,22 @@
 # Copyright 2019-present Kensho Technologies, LLC.
 """Utilities for rewriting IR to replace one set of locations with another."""
+from typing import Callable, Dict, TypeVar
+
 import six
 
+from ...compiler.expressions import ExpressionT
 from ..helpers import FoldScopeLocation, Location
+from ..metadata import QueryMetadataTable
+
+
+# This type exists in order to reduce the scope of what is allowed to be translated
+# since LocationT was not sufficiently specific.
+TranslatedLocationT = TypeVar("TranslatedLocationT", Location, FoldScopeLocation)
```

Comment:
If you do end up opening a `mypy` issue about this, please link to the issue here as a courtesy to the next person that tries to do the same thing :)